pkg/client/client.go

```diff
@@ -43,6 +48,42 @@ func New(cfg Config) (*CortexClient, error) {
 		"id":      cfg.ID,
 	}).Debugln("New ruler client created")
 
+	//if TLS flags, setup http client
+	if cfg.TLScaFile != "" && cfg.TLScertFile != "" && cfg.TLSkeyFile != "" {
+
+		// Load client cert
+		cert, err := tls.LoadX509KeyPair(cfg.TLScertFile, cfg.TLSkeyFile)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"tls-cert": cfg.TLScertFile,
+				"tls-key":  cfg.TLSkeyFile,
+			}).Fatal(err)
+		}
+
+		// Load CA cert
+		caCert, err := ioutil.ReadFile(cfg.TLScaFile)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"tls-ca": cfg.TLScaFile,
+			}).Fatal(err)
+		}
+		caCertPool := x509.NewCertPool()
+		caCertPool.AppendCertsFromPEM(caCert)
+
+		// Setup HTTPS client
+		tlsConfig := &tls.Config{
+			Certificates: []tls.Certificate{cert},
+			RootCAs:      caCertPool,
+		}
+		tlsConfig.BuildNameToCertificate()
+		transport := &http.Transport{TLSClientConfig: tlsConfig}
+		return &CortexClient{
```

Comment:
Instead of returning the client here, maybe its a good idea to simply set the http client. Something along the lines of: 

```golang
client := http.Client{}
if cfg.TLScaFile != "" && cfg.TLScertFile != "" && cfg.TLSkeyFile != "" {
  ...
  client = http.Client{Transport: transport}
}

return &CortexClient{
  key:      cfg.Key,
  id:       cfg.ID,
  endpoint: endpoint,
  client:   client,
}, nil
```

Reply:
great suggestion. thanks!